app/src/main/java/org/launchcode/g7/bookcollections/BookRecyclerViewAdapter.java

```diff
@@ -49,7 +49,7 @@ public void onClick(View v)
                 {
                     // Notify the active callbacks interface (the activity, if the
                     // fragment is attached to one) that an item has been selected.
-                    mListener.onListFragmentInteraction(holder.mItem);
+                    mListener.onListFragmentInteraction(holder.mItem,holder.getAdapterPosition());
```

Comment:
In order to get the pointer to the correct shelf in the file, the onListFragmentInteraction Event had to send the position along as a parameter.